### ISSUE TITLE
Use unambiguous JavaScript grammar to enforce strict mode

### ISSUE THREAD

**Issue by jdalton** [benjamn/reify#73]
Using ujsg you can detect when to add the "use strict" directive to better reflect esm behavior of implicit strict mode. It could also be used to do things like error on top level return statements and other esm only this if you wanted.

**Update:**
This will fix #112.

**Comment by jdalton** [comment]
This will fix #112.

**Comment by jdalton** [comment]
For reference for the ast to check:
```js
console.log(rootPath.stack[0].body && rootPath.stack[0].body[0]);
```
Logs
```
Node {
  type: 'ExpressionStatement',
  start: 10,
  end: 23,
  expression:
   Node {
     type: 'Literal',
     start: 10,
     end: 22,
     value: 'use strict',
     raw: '"use strict"' } }
```

**Comment by jdalton** [comment]
@benjamn Can you take a stab at adding this. I'm guessing it's a bit of AST sniffing and magic-string to insert `"use strict"` if it doesn't exist.

**Comment by jdalton** [comment]
Closed by #123.
